src/providers/oracle/qgsoracleprovider.cpp

```diff
@@ -1246,13 +1282,35 @@ bool QgsOracleProvider::addFeatures( QgsFeatureList &flist, QgsFeatureSink::Flag
 
     QStringList defaultValues;
     QList<int> fieldId;
+    QList<int> alwaysGenerated;
 
     if ( !mGeometryColumn.isNull() )
     {
       insert += quotedIdentifier( mGeometryColumn );
       values += '?';
       delim = ',';
     }
+    QString sql = QStringLiteral( "SELECT a.column_name "
```

Comment:
This looks very expensive to calculate on every feature insert. I think it should be retrieved once at provider initialization and remembered

Reply:
Indeed, good idea, it could be a class attribute ! Thanks, I will do that